### PR TITLE
CLI tool

### PR DESCRIPTION
**Pull request by osipov-mit**

### CODE REVIEW

File: tools/cli/src/gear/accounts.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      result[name] = keyring.addFromUri(seed) as KeyringPair;
+    } else if (isHex(seed)) {
+      result[name] = keyring.addFromSeed(hexToU8a(seed), {}, 'sr25519') as KeyringPair;
+    } else {
+      result[name] = keyring.addFromMnemonic(seed, {}, 'sr25519') as KeyringPair;
+    }
+    logger.info(`Set account ${name} to ${result[name].address}`, { lvl: 1 });
+  });
+  console.log();
+  return result;
```
Comment (Almazatun) [inline tools/cli/src/gear/accounts.ts:21]:
```suggestion
const keyringType = 'sr25519';
  const keyring = new Keyring({ type: keyringType });
  const result: Record<string, KeyringPair> = {};
  Object.entries(accounts).forEach(([name, seed]) => {
    if (seed.startsWith('//')) {
      result[name] = keyring.addFromUri(seed) as KeyringPair;
    } else if (isHex(seed)) {
      result[name] = keyring.addFromSeed(hexToU8a(seed), {}, keyringType) as KeyringPair;
    } else {
      result[name] = keyring.addFromMnemonic(seed, {}, keyringType) as KeyringPair;
    }
    logger.info(`Set account ${name} to ${result[name].address}`, { lvl: 1 });
  });
  console.log();
  return result;
```

File: tools/cli/src/gear/code.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  if (uploadedCode.isSome) {
+    const blockHash = await api.blocks.getBlockHash(uploadedCode.unwrap().blockNumber);
+    logger.warn(`Code is already uploaded in block ${blockHash}`, { lvl: 1 });
+    return { codeHash, blockHash: blockHash.toHex() };
+  }
+
+  await api.code.upload(code);
+
+  const blockHash: `0x${string}` = await new Promise((resolve, reject) =>
```
Comment (Almazatun) [inline tools/cli/src/gear/code.ts:25]: Would be good to create for this `0x${string}` custom type. Because you have used the type also in other files too.

**Review by Almazatun — APPROVED** [review]

**Review by AndrePanin — APPROVED** [review]

File: tools/cli/src/gear/findEvents.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      const reply = await new Promise((resolve) => {
+        if (event.data.message.source.eq(programId)) {
+          if (event.data.message.details.isSome) {
+            if (event.data.message.details.unwrap().isReply) {
+              if (event.data.message.details.unwrap().asReply.replyTo.eq(messageId)) {
+                resolve([event.data.message.id.toHex(), event.data.message.payload]);
+              }
+            }
+          }
+        }
```
[the remark this answers is not in the document]
Comment (osipov-mit) [reply tools/cli/src/gear/findEvents.ts]: I don't see point of doing it, but i've refactored this part for a shorter view

File: tools/cli/src/gear/accounts.ts (hunk 1 above)
Comment (osipov-mit) [reply tools/cli/src/gear/accounts.ts]: agree

**Review by nikitayutanov — APPROVED** [review]

**Pull request merged**